l10n_cn_account_voucher/models/account_move.py

```diff
@@ -0,0 +1,132 @@
+        Exchange rate: Debit or Credit / currency ammount
+        Why not get it from currency code + date ?
+        '''
+        exchange_rate = False
+        if line.amount_currency:
+            if line.debit > 0:
+                exchange_rate = line.debit / line.amount_currency
+            if line.credit > 0:
+                exchange_rate = line.credit / (-1 * line.amount_currency)
+        return round(exchange_rate, 6)
+
+    def _get_unit_price(self, line):
+        '''
+        Unit price：Debit or Credit / Quantity
+        '''
+        unit_price = False
+        if line.quantity:
+            if line.debit > 0:
+                unit_price = line.debit / line.quantity
+            if line.credit > 0:
+                unit_price = line.credit / line.quantity
+        return unit_price
+
+    def _rmb_format(self, value):
+        """
+        Separate numerical according to the figures
+        """
+        rounding = self.env['res.currency'].browse(
+            self.company_id.currency_id.id).rounding
+
+        if value < rounding:
+            # if value is 0，return 0.00
+            return (['' for i in range(12)] + list((
+                '%0.2f' % 0).replace('.', '')))[-12:]
+        # change num to string, remove The decimal point
+        # get 12 itmes and return
+        return (['' for i in range(12)] + list((
+            '%0.2f' % value).replace('.', '')))[-12:]
+
+    def _rmb_upper(self, value):
+        """
+        in capital letters
+        from：http://topic.csdn.net/u/20091129/20/b778a93d-9f8f-4829-9297-d05b08a23f80.html
+        To float value is returned, unicode string
+        """
+        rmbmap = [u"零", u"壹", u"贰", u"叁", u"肆", u"伍", u"陆", u"柒", u"捌", u"玖"]
+        unit = [u"分", u"角", u"元", u"拾", u"佰", u"仟", u"万", u"拾", u"佰", u"仟",
+                u"亿", u"拾", u"佰", u"仟", u"万", u"拾", u"佰", u"仟", u"兆"]
+        nums = map(int, list(str('%0.2f' % value).replace('.', '')))
+        words = []
+        # Tag 0 times in a row, to remove a swastika,
+        # insert zero word or timely
+        zflag = 0
+        start = len(nums) - 3
+        # Make I corresponds to the actual figures,
+        # negative for the corner points
+        for i in range(start, -3, -1):
+            if 0 != nums[start - i] or len(words) == 0:
+                if zflag:
+                    words.append(rmbmap[0])
+                    zflag = 0
+                words.append(rmbmap[nums[start - i]])
+                words.append(unit[i + 2])
+            # Control 'ten thousand/yuan'
+            elif 0 == i or (0 == i % 4 and zflag < 3):
+                words.append(unit[i + 2])
+                zflag = 0
+            else:
+                zflag += 1
+        # At the end of the 'points' to fill the whole word
+        if words[-1] != unit[0]:
+            words.append(u"整")
+        return ''.join(words)
+
+
+class AccountMoveLine(models.Model):
+    _inherit = 'account.move.line'
+
+    currency_rate = fields.Float(
```

Comment:
This will decrease the performance being not stored. Any reason for not storing?